spring-integration/spring-boot-autoconfigure/src/main/java/ai/timefold/solver/spring/boot/autoconfigure/config/SolverProperties.java

```diff
@@ -1,12 +1,24 @@
 package ai.timefold.solver.spring.boot.autoconfigure.config;
 
+import java.util.Map;
+import java.util.Set;
+
 import ai.timefold.solver.core.api.domain.common.DomainAccessType;
 import ai.timefold.solver.core.api.score.stream.ConstraintStreamImplType;
 import ai.timefold.solver.core.config.solver.EnvironmentMode;
 
 import org.springframework.boot.context.properties.NestedConfigurationProperty;
 
 public class SolverProperties {
+    public static final Set<String> VALID_FIELD_NAMES_SET =
+            Set.of("solver-config-xml", "environment-mode", "daemon", "move-thread-count", "domain-access-type",
+                    "constraint-stream-impl-type", "termination");
```

Comment:
It is a good idea to have a constant for this.
However, I'd go further and have constants even for the names themselves - that way, you don't need to repeat them later in this class.